### ISSUE TITLE
Firedrake-update confusion

### ISSUE THREAD

**Issue by wence-** [firedrakeproject/firedrake#690]
Consider the following sequence:

```
$ python firedrake-install
$ . firedrake/bin/activate
$ firedrake-update
...
Installing virtualenv.
Collecting virtualenv
  Using cached virtualenv-14.0.5-py2.py3-none-any.whl
Installing collected packages: virtualenv
Successfully installed virtualenv-14.0.5
Virtual env installed. Please run firedrake-install again.
$ firedrake-install
...
Creating firedrake virtualenv.
Please use the *system* python to run this script
Traceback (most recent call last):
  File "/data/lmitche1/fd-install-tests/firedrake/bin/firedrake-install", line 665, in <module>
    virtualenv.create_environment("firedrake", site_packages=False)
  File "/data/lmitche1/fd-install-tests/firedrake/local/lib/python2.7/site-packages/virtualenv.py", line 884, in create_environment
    site_packages=site_packages, clear=clear, symlink=symlink))
  File "/data/lmitche1/fd-install-tests/firedrake/lib/python2.7/posixpath.py", line 360, in abspath
    if not isabs(path):
  File "/data/lmitche1/fd-install-tests/firedrake/lib/python2.7/posixpath.py", line 54, in isabs
    return s.startswith('/')
AttributeError: 'NoneType' object has no attribute 'startswith'
```

Now, we shouldn't need to run `firedrake-install` after running `firedrake-update`.  But that's what the script says.

I think the symptom of this is that inside the activated virtualenv `virtualenv` is not installed, and so we hit this code in `firedrake-update`
```python
elif not (args.system or args.user or args.prefix):
    try:
        import virtualenv
    except:
        stdout.write("Installing virtualenv.\n")
        # sys.real_prefix exists if we are already in the virtualenv
        if hasattr(sys, "real_prefix"):
            check_call(["pip", "install", "virtualenv"])
        else:
            check_call(["pip", "install", "--user", "virtualenv"])
        quit("Virtual env installed. Please run firedrake-install again.\n")
```

And at this point, running `firedrake-install` then fails as above.

I think we should probably install `virtualenv` inside the virtualenv.  Or else do some error checking to see how we're being run and provide a correct message.

**Comment by wence-** [comment]
The message confusion part of this was fixed in #693.  We should probably still just install virtualenv inside the virtualenv when running firedrake-install for the first time.

**Comment by miklos1** [comment]
I don't think that we actually need virtualenv inside the virtualenv. Why don't we just push `try: import virtualenv; expect: ...` into `if mode == 'install':` in the install script?

**Comment by wence-** [comment]
Let me try that.

**Comment by wence-** [comment]
Yes, that seems to work, let's wait for a test of #698.

**Comment by wence-** [comment]
This is now fixed by #698.
